content/setup-ubuntu-linux-for-laravel-developer.md

```diff
@@ -0,0 +1,123 @@
+sudo apt install php7.4-common php7.4-mysql php7.4-xml php7.4-xmlrpc php7.4-curl php7.4-gd php7.4-imagick php7.4-cli php7.4-dev php7.4-imap php7.4-mbstring php7.4-opcache php7.4-soap php7.4-zip php7.4-intl -y
+```
+
+# Composer
+
+Kemudian pasang composer
+
+```bash
+php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+php -r "if (hash_file('sha384', 'composer-setup.php') === '906a84df04cea2aa72f40b5f787e49f22d4c2f19492ac310e8cba5b96ac8b64115ac402c8cd292b8a03482574915d1a8') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
+php composer-setup.php
+php -r "unlink('composer-setup.php');"
+```
+
+Setelah proses diatas sudah dijalankan satu per satu, pindahkan composer.phar ke direktori PATH.
+
+```bash
+sudo mv composer.phar /usr/local/bin/composer
+```
+
+# Node.js
+
+Kita perlu memasang Node.js untuk memasang NPM (Node Package Manager), caranya pun cukup mudah. Jalankan perintah di bawah ini satu per satu:
+
+```bash
+curl -sL https://deb.nodesource.com/setup_17.x -o nodesource_setup.sh
+sudo bash nodesource_setup.sh
+sudo apt install nodejs
+```
+
+Untuk version pada `https://deb.nodesource.com/setup_17.x`, ini opsional ya. Kalian bisa menyesuaikan sesuai dengan kebutuhan.
+
+# MySQL
+
+Nah, untuk MySQL ini kalian sebenarnya tidak perlu bingung. Cukup buka link ini
+
+- [`MySQL Downloads`](https://dev.mysql.com/downloads/mysql/),
+
+Download MySQL-nya dan sesuaikan dengan Distro OS Linux kalian. Kemudian ikuti langkah-langkahnya seperti yang ada di dokumentasi ini
+
+- [`Cara Install`](https://dev.mysql.com/doc/refman/8.0/en/linux-installation-debian.html).
+
+Lohhh, kok ke dokumentasi? Ya sekalian belajar baca-baca dokumentasinya juga hehehe :). Belajar dari yang mudah untuk diikuti terlebih dahulu aja okay ^~^.
+
+Untuk MySQL Client nya sendiri, saya menggunakan ekstensi Database yang ada di Visual Studio Code. Jadi, saya ga perlu install PHPMyAdmin, MySQL Client, atau sejenisnya.
+
+Oiya, disini saya tidak menggunakan Web Server ya. Karena Laravel sendiri sudah menyediakannya dengan cara menjalankan command `php artisan serve`.
+
+Sip, selanjutnya kita install Go untuk kebutuhan kalau kita pengen ngoding Go.
+
+# Go
+
+Sebenarnya, langkahnya cukup mudah. Pertama [Download Go disini](https://go.dev/doc/install), setelah itu jalankan perintah dibawah ini satu per satu
+
+```bash
+rm -rf /usr/local/go && tar -C /usr/local -xzf go1.17.6.linux-amd64.tar.gz
+export PATH=$PATH:/usr/local/go/bin
+```
+
+Untuk catatan, terkadang `PATH` ini tidak bersifat permanen ya. Apalagi kalau kalian menggunakan `ZSH` sebagai default terminal kalian di Linux. Kalian perlu menambahkan konfigurasi export-nya secara manual dengan cara,
+
+```bash
+sudo nano ~/.zshrc
+```
+
+Kemudian tambahkan kode bash seperti dibawah ini, jangan lupa disimpan ya.
+
+```bash
+export PATH=$PATH:/usr/local/go/bin
+```
+
+Udah deh, tinggal cek. Apakah sudah terpasang atau belum dengan command
```

Comment:
Masih perlu `source ~/.bashrc` atau config file shell yang dipakai sebelum lanjut.

Reply:
hee iyakah? di aku kemarin cuma itu aja ki?

Reply:
restart shellnya atau jalanin `source ~/.bashrc` atau `source ~/.zshrc`

Reply:
maish bisa kok :/

Reply:
https://user-images.githubusercontent.com/51877647/151927128-7682f69e-d026-49db-b659-0452beefb781.mp4

Reply:
ohhhh brti cukup tambah `bash` buat restart nya gitu aja?

Reply:
Itu berarti numpuk bash diatas bash. Saranku pakai yg source

Reply:
abis source, terus exit gtu?.


soale saya kemrin engga ngelakuin itu. jadi ngawang2 :"

Reply:
Exit?? Tutup terminal gitu?

Reply:
ngga, kalo exit itu restart shellnya
pake `source` kalo gamau restart shellnya

Reply:
yes, brti ini bahasa nya restart shell atau apa yak? wkwkwk

Reply:
Tulis aja restart terminal. Karena ini how to nya kadang orang nggak mau susah susah. Banyak beginner yg baca. Klo bliding edge targetnya ya tulis source wkwkwk

Reply:
sip, udah
https://github.com/teknologi-umum/blog/compare/master...WahidinAji:master#diff-1bfd9fdd83478db0cc7289801f7662a81a84dd131bae5df29d6e5de79d840f18R102-R108